draft-thomson-quic-tls.md

```diff
@@ -533,16 +555,21 @@ Issue:
 : Is it possible to send a `STREAM` frame for stream 1 that contains no data?
   Is this detectable?  Does it comprise an attack?  For instance, could an
   attacker inject a frame that appears to contain TLS application data?
-
+  TODO: Currently we consider it an error to send a stream frame with no data
+  and no fin bit set. (And fins are prohibited on stream 1).
 
 ### ACK Frames
 
 `ACK` frames are permitted prior to the handshake being complete.  However, an
 unauthenticated `ACK` frame can only be used to obtain NACK ranges.  Timestamps
 MUST NOT be included in an unprotected ACK frame, since these might be modified
 by an attacker with the intent of altering congestion control response.
-Information on FEC-revived packets is redundant, since use of FEC in this phase
-is prohibited.
+TODO: I don't think this is a good restriction. Before crypto is established,
+receive timestamps help set the retransmission timers to the rigth value. You're
+totally right that an attacker can mess with them, but I don't think that we
+should totally throw out the value here, if we can avoid it. That being said,
+I think that considering a packet received (which wasn't actually received)
+is actually worse than a confused RTT estimate :/
```

Comment:
Let's leave this as an open issue.  I think that we will ultimately conclude that ACKs are necessary and we will have to accept the risks.  And you are right that timing is useful.

What I really want is to train in an RTT value and learn about what a peer has without having that value influence the secured values.  Maybe we could let cleartext ACKs do anything at all.  Accept the risks that come with that, but start a second RTT measurement once we go fully secure (we can't use 0-RTT because replay is a big deal with 0-RTT).  Then once that second set of values is spun up, switch over to using it and throw out the tainted data.  You could even transition from one to the other to improve reliability.

At least that way you can't mess with the connection perpetually by injecting a few ACK packets from off-path.

I have a new section that I think is better.